src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -451,6 +451,71 @@ begin
   cyclotomic'_eq_X_pow_sub_one_div hpos hz, finset.prod_congr (refl k.proper_divisors) h]
 end
 
+lemma eq_cyclotomic_iff {R : Type*} [comm_ring R] [nontrivial R] {n : ℕ} (hpos: 0 < n)
+  (P : polynomial R) :
+  P = cyclotomic n R ↔ P * (∏ i in nat.proper_divisors n, polynomial.cyclotomic i R) = X ^ n - 1 :=
+begin
+  split,
+  { intro hcycl,
+    rw [hcycl, ← finset.prod_insert (@nat.proper_divisors.not_self_mem n),
+      ← nat.divisors_eq_proper_divisors_insert_self_of_pos hpos],
+    exact prod_cyclotomic_eq_X_pow_sub_one hpos R },
+  { intro hP,
+    have prod_monic : (∏ i in nat.proper_divisors n, cyclotomic i R).monic,
+    { apply monic_prod_of_monic,
+      intros i hi,
+      exact cyclotomic.monic i R },
+    rw [@cyclotomic_eq_X_pow_sub_one_div R _ _ _ hpos,
+      (div_mod_by_monic_unique P 0 prod_monic _).1],
+    split,
+    { rwa [zero_add, mul_comm] },
+    rw [degree_zero, bot_lt_iff_ne_bot],
+    intro h,
+    exact monic.ne_zero prod_monic (degree_eq_bot.1 h) },
+end
+
+/-- If `p` is prime, then `cyclotomic p R = geom_series X p`. -/
+lemma cyclotomic_of_prime {R : Type*} [comm_ring R] [nontrivial R] {p : ℕ} (hp : nat.prime p) :
+  geom_series X p = cyclotomic p R :=
```

Comment:
This statement doesn't match the docstring (sides are swapped).
Also, I would use the name `cyclotomic_eq_geom_series` (the `of_prime` can be left out, most users will know that the result requires that hypothesis).

Reply:
Done it.